src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement(CREATE_SQL, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Expected to insert at list one row"
+                        + "but were inserted zero rows");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t create new book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Book book = new Book();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_BY_ID_SQL)) {
+            statement.setObject(1, id, Types.BIGINT);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                book = collectBook(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t find book with id = " + id, e);
+        }
+        return Optional.of(book);
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> bookList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_ALL_SQL)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                bookList.add(collectBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t find all books from DB " + e);
+        }
+        return bookList;
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(UPDATE_BY_ID_SQL)) {
+            statement.setObject(1, book.getTitle(), Types.VARCHAR);
+            statement.setObject(2, book.getPrice(), Types.DECIMAL);
+            statement.setObject(3, book.getId(), Types.BIGINT);
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("At least one parameter must be update,"
+                        + "but anyone was`nt updated");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update data in DB " + e);
```

Comment:
Similar to the previous comment, avoid concatenating the exception message with `e`. Pass `e` as the second argument to the `DataProcessingException` constructor.